### DISCUSSION TITLE
Questions about JS and browser support

### DISCUSSION THREAD

**Discussion by nskillen** [nzbgetcom/nzbget#278] [General]
I have a few ideas for improvements to the server page I'd like to implement, mainly centered around the ability to collapse server blocks I'm not interested in at the moment.

While I worked through understanding the way the current code works, I noticed that there's nothing like arrow functions (`(arg) => { /* body */ }`) or template literals (`` `string ${variable}` ``) currently being used.

Is this because it will break things on some supported devices/browsers? Should I avoid them in any PRs that I end up submitting?

**Comment by dnzbk** [comment]
Yes, I wouldn't recommend using such ES6 features right now. 
I'm thinking of redesigning webui to make it look modern and user friendly and for that I will use technologies like TypeScript and ReactJS, which will simplify development, maintenance and of course resolve such problems.

**Comment by nskillen** [comment]
Thanks. Maybe I'll take a look at what might be needed to set up react instead. Were you leaning towards a full react app, or something lightweight like nano-react-app? The ecosystem is a lot bigger than it was the first time I looked at react...

**Comment by dnzbk** [comment]
I haven't heard about nano-react but I'll consider using that library too, thanks.
